binding/client.go

```diff
@@ -775,13 +775,16 @@ func (r *Client) restError(response *http.Response) (err error) {
 	switch status {
 	case http.StatusConflict:
 		restError := &Conflict{}
-		err = restError.With(response)
+		restError.With(response)
+		err = restError
 	case http.StatusNotFound:
 		restError := &NotFound{}
-		err = restError.With(response)
+		restError.With(response)
+		err = restError
 	default:
 		restError := &RestError{}
-		err = restError.With(response)
+		restError.With(response)
+		err = restError
 	}
```

Comment:
**hint**: RestError.With() returned a RestError not the concrete type.  It was _object slicing_.  As a result, restError() was not returned the concrete type as intended.